### PR TITLE
make openstack provider work on ovh cloud

### PR DESCRIPTION
**Pull request by tlamr**
**What this PR does / why we need it**:
In this PR we workaround Opestack limitation in OVH Cloud. They do not allow to call GET regions endpoint.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
NONE
```

cc: @ovh @cbourgois

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
@tlamr thank you for the contribution. Can you please ensure that your commits are signed off? 
https://github.com/kubermatic/machine-controller/blob/master/CONTRIBUTING.md#certificate-of-origin

**Comment by tlamr** [comment]
Hello @ahmedwaleedmalik, just signed them off, thanks for the notice :)

**Comment by ahmedwaleedmalik** [comment]
@tlamr can you please take care of the failures? Mostly linting errors.

File: pkg/cloudprovider/provider/openstack/provider.go [hunk 1]
```diff
@@ -439,6 +439,7 @@ func (p *provider) AddDefaults(spec clusterv1alpha1.MachineSpec) (clusterv1alpha
 	return spec, nil
 }
 
+//nolint:gocyclo
```
Comment (ahmedwaleedmalik) [inline pkg/cloudprovider/provider/openstack/provider.go:442]: This is not the appropriate fix for this. Please add an exception for this in `golangci.yaml`.

Let me know if you need assistance with that

File: pkg/cloudprovider/provider/openstack/provider.go [hunk 2]
```diff
@@ -480,8 +481,10 @@ func (p *provider) Validate(spec clusterv1alpha1.MachineSpec) error {
 	}
 
 	// Required fields
-	if _, err := getRegion(client, c.Region); err != nil {
-		return fmt.Errorf("failed to get region %q: %v", c.Region, err)
+	if !strings.Contains(c.IdentityEndpoint, "auth.cloud.ovh.net") {
```
Comment (ahmedwaleedmalik) [inline pkg/cloudprovider/provider/openstack/provider.go:484]: Please create a const for "auth.cloud.ovh.net" and move it to the top of the file. That will help in visibility.

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: pkg/cloudprovider/provider/openstack/provider.go (hunk 1 above)
Comment (tlamr) [reply pkg/cloudprovider/provider/openstack/provider.go:442]: ok, thanks! :)

File: pkg/cloudprovider/provider/openstack/provider.go (hunk 2 above)
Comment (tlamr) [reply pkg/cloudprovider/provider/openstack/provider.go]: ok, thanks! :)

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**